backend/qemu.pm

```diff
@@ -201,22 +201,54 @@ sub can_handle {
     return;
 }
 
+sub open_file_and_send_fd_to_qemu {
+    my ($self, $path, $fdname) = @_;
+    my $rsp;
+
+    my $fd = POSIX::open($path, &POSIX::O_CREAT | &POSIX::O_RDWR);
+    die "Failed to open $path: $!" unless (defined $fd);
+
+    $rsp = $self->handle_qmp_command(
+        {execute => 'getfd', arguments => {fdname => $fdname}},
+        send_fd => $fd,
+        fatal   => 1
+    );
+    POSIX::close($fd);
+}
+
 sub save_memory_dump {
     my ($self, $args) = @_;
-    my $rsp = 0;
+    my $filename = $args->{filename} . '-vm-memory-dump';
+    my $fdname   = 'dumpfd';
+    my $vars     = \%bmwqemu::vars;
 
-    bmwqemu::diag("Migrating the machine.");
+    my $rsp = $self->handle_qmp_command({execute => 'query-status'}, fatal => 1);
+    bmwqemu::diag("Migrating the machine (Current VM state is $rsp->{return}->{status}).");
+    my $was_running = $rsp->{return}->{status} eq 'running';
+
+    # Ensure slow dump times are not due to a transfer rate cap
+    $rsp = $self->handle_qmp_command(
+        {
+            execute   => 'migrate-set-parameters',
+            arguments => {'max-bandwidth' => $vars->{QEMU_MAX_BANDWIDTH} || 2147483648}
```

Comment:
where does this magic number come from? Better have a constant with an explanation?